### PR TITLE
Adding the build step for the JS bundle

### PR DESCRIPTION
**Pull request by pdobrev**
Several changes that (somewhat unfortunately) got bundled up in the same PR since they were required to get it working.

* Got the JS part of pipeable (called SophiaJS internally) into the framework and added it as a build step. 
It's implemented in TypeScript and compiled down to JS, the sources live in `Sources/sophiajs`. The CI now builds the TypeScript file and copies it to the iOS bundle, where the framework can load it.

* There was quite a bit of struggling getting the framework to find its own Resource (the built "sophia.js" file)
The problem was that as a Swift package, it works in a given way (`Bundle.module.path()`), but for that to work, you need to compile it using the Swift Package Manager and not the XCode project. If you build it using the XCode project, then you don't have `Bundle.module`. But if you include it as a Swift package, then that's the only option that works. Overall, it's quite the mess, so for now we've solved it as follows: there's a XC Workspace which includes both the framework and the sample app. The sample app requires the framework as a framework and not as a swift package. And everything works. When it comes to publishing, we'll need a separate build pipeline for Swift Packages, but it is what it is.

### CODE REVIEW

File: .github/workflows/ios-build-test.yml [hunk 1]
```diff
@@ -15,6 +15,16 @@ jobs:
       - name: checkout repository
         uses: actions/checkout@v3
 
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 18
```
Comment (ibratoev) [inline .github/workflows/ios-build-test.yml:20]: Consider locking the node version in a `.nvmrc` file at the top of the repo and using `node-version-file`. That way other tools can setup the correct node version automatically.

File: .github/workflows/ios-build-test.yml [hunk 2]
```diff
@@ -15,6 +15,16 @@ jobs:
...
+          cache: 'npm'
```
Comment (ibratoev) [inline .github/workflows/ios-build-test.yml:21]: Btw, from my experience sometimes caching is slower than not caching npm packages... Consider simplifying it and not caching.

**Review by ibratoev — APPROVED** [review]
Added a few non critical comments.

File: .github/workflows/ios-build-test.yml (hunk 1 above)
Comment (pdobrev) [reply .github/workflows/ios-build-test.yml:20]: Sounds good, I'll do that, thanks!

File: .github/workflows/ios-build-test.yml (hunk 2 above)
Comment (pdobrev) [reply .github/workflows/ios-build-test.yml:21]: OK, we can skip for now and measure; I'll do 2 runs to see which one is faster and if there's no difference, will remove. There's relatively few packages now anyway.

Comment (pdobrev) [reply .github/workflows/ios-build-test.yml]: Right. With caching it's 4s (cache unpack, npm ci, cache pack) vs 3s or 2s. So I'll remove it. The time is negligible anyway.

**Pull request merged**